### PR TITLE
Add test which reproduces bug

### PR DESCRIPTION
**Pull request by Mykhailenko**
This PR adds TestRMReconnectioWhileRunning which reproduces [this](https://trello.com/c/3GUVeOLK/2828-stuck-scheduler-10-cannot-send-request-terminate-to-body) bug. Now it is marked as ignore because it fails.

### CODE REVIEW

File: scheduler/scheduler-server/src/test/java/functionaltests/rm/TestRMReconnectioWhileRunning.java [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+        schedulerHelper = new SchedulerTHelper(false, true);
+        schedulerHelper.createNodeSource("local", 3);
+        schedulerHelper.getSchedulerInterface().changePolicy("functionaltests.rm.PolicyWhichThrowsExceptions");
+        JobId jobId = schedulerHelper.submitJob(new File(runningJob.toURI()).getAbsolutePath());
+        JobId jobId2 = schedulerHelper.submitJob(new File(runningJob1.toURI()).getAbsolutePath());
+        JobId jobId3 = schedulerHelper.submitJob(new File(runningJob2.toURI()).getAbsolutePath());
+
+        schedulerHelper.waitForEventTaskRunning(jobId, "running_task_for20s");
+
+        Thread.sleep(200000);
```
Comment (sonarqube-activeeon) [inline scheduler/scheduler-server/src/test/java/functionaltests/rm/TestRMReconnectioWhileRunning.java]: MAJOR Remove this use of "Thread.sleep()". [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2925)

File: rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/core/RMCore.java [hunk 2]
```diff
@@ -1929,17 +1930,30 @@ public BooleanWrapper disconnect() {
...
                     if (rmnode.isToRemove()) {
+                        logger.info("RMCore: rmnode.isToRemove()");
                         removeNodeFromCoreAndSource(rmnode, client);
                     } else if (rmnode.isBusy()) {
+                        logger.info("RMCore: rmnode.isBusy()");
                         nodesToRelease.add(rmnode);
                     }
+                }else{
+                    logger.info("RMCore not going to enter: " + rmnode.getNodeName() + "; " + client + "; " + rmnode.getOwner() + "; " + clientId + "; " + rmnode.getOwner().getId());
+                    // local-TEST-0_1;
```
Comment (sonarqube-activeeon) [inline rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/core/RMCore.java]: MAJOR This block of commented-out lines of code should be removed. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ACommentedOutCodeLine)

File: rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/core/RMCore.java [hunk 3]
```diff
@@ -1929,17 +1930,30 @@ public BooleanWrapper disconnect() {
     public void disconnect(UniqueID clientId) {
```
Comment (sonarqube-activeeon) [inline rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/core/RMCore.java]: CRITICAL Refactor this method to reduce its Cognitive Complexity from 16 to the 15 allowed. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)

**Comment by sonarqube-activeeon** [comment]
SonarQube analysis reported 3 issues
* ![CRITICAL][CRITICAL] 1 critical
* ![MAJOR][MAJOR] 2 major

Watch the comments in this conversation to review them.

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: scheduler/scheduler-server/src/test/java/functionaltests/rm/TestRMReconnectioWhileRunning.java [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+            SchedulerTHelper.log("Killing previous scheduler.");
+            TestScheduler.kill();
+        }
+        initConfigs();
+    }
+
+
+    @Ignore
+    @Test
+    public void test() throws Exception {
```
Comment (sonarqube-activeeon) [inline scheduler/scheduler-server/src/test/java/functionaltests/rm/TestRMReconnectioWhileRunning.java]: MAJOR Fix or remove this skipped unit test [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1607)

**Comment by sonarqube-activeeon** [comment]
SonarQube analysis reported 2 issues
* ![MAJOR][MAJOR] 2 major

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: scheduler/scheduler-server/src/test/java/functionaltests/rm/PolicyWhichThrowsExceptions.java [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+    static final Logger logger = Logger.getLogger(PolicyWhichThrowsExceptions.class);
+
+    private int counter = 0;
+
+
+    @Override
+    public boolean isTaskExecutable(NodeSet selectedNodes, EligibleTaskDescriptor task) {
+        logger.info("PolicyWhichThrowsExceptions::isTaskExecutable is called " + counter);
+        if(counter == 1 || counter == 2 || counter == 3 || counter == 4){
```
Comment (marcocast) [inline scheduler/scheduler-server/src/test/java/functionaltests/rm/PolicyWhichThrowsExceptions.java]: if(counter > 0 && counter <= 4){

File: scheduler/scheduler-server/src/test/java/functionaltests/rm/TestRMReconnectioWhileRunning.java (hunk 4 above)
Comment (marcocast) [reply scheduler/scheduler-server/src/test/java/functionaltests/rm/TestRMReconnectioWhileRunning.java]: As sonar said, fix or remove the @ignore test

File: scheduler/scheduler-server/src/test/java/functionaltests/rm/TestRMReconnectioWhileRunning.java [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+        schedulerHelper = new SchedulerTHelper(false, true);
+        schedulerHelper.createNodeSource("local", 3);
+        schedulerHelper.getSchedulerInterface().changePolicy("functionaltests.rm.PolicyWhichThrowsExceptions");
+        JobId jobId = schedulerHelper.submitJob(new File(runningJob.toURI()).getAbsolutePath());
+        JobId jobId2 = schedulerHelper.submitJob(new File(runningJob1.toURI()).getAbsolutePath());
+        JobId jobId3 = schedulerHelper.submitJob(new File(runningJob2.toURI()).getAbsolutePath());
+
+        schedulerHelper.waitForEventTaskRunning(jobId, "running_task_for20s");
+
+        Thread.sleep(200000);
```
Comment (marcocast) [inline scheduler/scheduler-server/src/test/java/functionaltests/rm/TestRMReconnectioWhileRunning.java]: why this sleep?

File: scheduler/scheduler-server/src/test/java/functionaltests/rm/PolicyWhichThrowsExceptions.java (hunk 5 above)
Comment (Mykhailenko) [reply scheduler/scheduler-server/src/test/java/functionaltests/rm/PolicyWhichThrowsExceptions.java]: ok.

File: scheduler/scheduler-server/src/test/java/functionaltests/rm/TestRMReconnectioWhileRunning.java (hunk 6 above)
Comment (Mykhailenko) [reply scheduler/scheduler-server/src/test/java/functionaltests/rm/TestRMReconnectioWhileRunning.java]: To make sure that I waited enough that job could finish.

File: scheduler/scheduler-server/src/test/java/functionaltests/rm/TestRMReconnectioWhileRunning.java (hunk 4 above)
Comment (Mykhailenko) [reply scheduler/scheduler-server/src/test/java/functionaltests/rm/TestRMReconnectioWhileRunning.java]: If I remove this test will fail constantly, do we want it? I mean I dont have idea how to fix it yet

File: scheduler/scheduler-server/src/test/java/functionaltests/rm/TestRMReconnectioWhileRunning.java [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+        schedulerHelper = new SchedulerTHelper(false, true);
+        schedulerHelper.createNodeSource("local", 3);
+        schedulerHelper.getSchedulerInterface().changePolicy("functionaltests.rm.PolicyWhichThrowsExceptions");
+        JobId jobId = schedulerHelper.submitJob(new File(runningJob.toURI()).getAbsolutePath());
+        JobId jobId2 = schedulerHelper.submitJob(new File(runningJob1.toURI()).getAbsolutePath());
+        JobId jobId3 = schedulerHelper.submitJob(new File(runningJob2.toURI()).getAbsolutePath());
+
+        schedulerHelper.waitForEventTaskRunning(jobId, "running_task_for20s");
+
+        Thread.sleep(200000); // sleep enough that all jobs should be able to finish
```
Comment (sonarqube-activeeon) [inline scheduler/scheduler-server/src/test/java/functionaltests/rm/TestRMReconnectioWhileRunning.java]: MAJOR Remove this use of "Thread.sleep()". [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2925)

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/rmproxies/RMProxiesManager.java [hunk 8]
```diff
@@ -102,7 +102,7 @@ public RMProxiesManager(Credentials schedulerProxyCredentials) throws RMExceptio
      *
      * @param rmURI The URI of a started Resource Manager
      */
-    public abstract void rebindRMProxiesManager(final URI rmURI) throws RMException, RMProxyCreationException;
+    public abstract boolean rebindRMProxiesManager(final URI rmURI) throws RMException, RMProxyCreationException;
```
Comment (sonarqube-activeeon) [inline scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/rmproxies/RMProxiesManager.java]: MINOR Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/rmproxies/SingleConnectionRMProxiesManager.java [hunk 9]
```diff
@@ -52,9 +52,9 @@ synchronized public URI getRmUrl() {
     }
 
     @Override
-    synchronized public void rebindRMProxiesManager(URI rmURI) throws RMException, RMProxyCreationException {
+    synchronized public boolean rebindRMProxiesManager(URI rmURI) throws RMException, RMProxyCreationException {
```
Comment (sonarqube-activeeon) [inline scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/rmproxies/SingleConnectionRMProxiesManager.java]: MINOR Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/rmproxies/PerUserConnectionRMProxiesManager.java [hunk 10]
```diff
@@ -66,14 +66,14 @@ public URI getRmUrl() {
     }
 
     @Override
-    public void rebindRMProxiesManager(URI rmURI) throws RMException, RMProxyCreationException {
+    public boolean rebindRMProxiesManager(URI rmURI) throws RMException, RMProxyCreationException {
```
Comment (sonarqube-activeeon) [inline scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/rmproxies/PerUserConnectionRMProxiesManager.java]: MINOR Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/descriptor/JobDescriptorImpl.java [hunk 11]
```diff
@@ -240,6 +240,16 @@ public void reStart(TaskId taskId) {
         eligibleTasks.put(taskId, (EligibleTaskDescriptor) runningTasks.remove(taskId));
     }
 
+    synchronized public void restartAllRunningTasls() {
```
Comment (sonarqube-activeeon) [inline scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/descriptor/JobDescriptorImpl.java]: MINOR Reorder the modifiers to comply with the Java Language Specification. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AModifiersOrderCheck)

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/rmproxies/RMProxy.java [hunk 12]
```diff
@@ -98,12 +98,12 @@ public RMState getState() {
         return proxyActiveObject.getState();
     }
 
-    public void rebind(URI rmURI) throws RMException, RMProxyCreationException {
+    public boolean rebind(URI rmURI) throws RMException, RMProxyCreationException {
```
Comment (sonarqube-activeeon) [inline scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/rmproxies/RMProxy.java]: MINOR Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)

**Comment by sonarqube-activeeon** [comment]
SonarQube analysis reported 19 issues
* ![CRITICAL][CRITICAL] 1 critical
* ![MAJOR][MAJOR] 1 major
* ![MINOR][MINOR] 17 minor

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [InternalJob.java#L399](https://github.com/ow2-proactive/scheduling/blob/0ce8cc469c23b4d53489c8b913c281664c91b6e0/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/job/InternalJob.java#L399): Add a default case to this switch. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ASwitchLastCaseIsDefaultCheck)
1. ![MINOR][MINOR] [LiveJobs.java#L705](https://github.com/ow2-proactive/scheduling/blob/0ce8cc469c23b4d53489c8b913c281664c91b6e0/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/LiveJobs.java#L705): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. ![MINOR][MINOR] [LiveJobs.java#L793](https://github.com/ow2-proactive/scheduling/blob/0ce8cc469c23b4d53489c8b913c281664c91b6e0/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/LiveJobs.java#L793): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. ![MINOR][MINOR] [LiveJobs.java#L848](https://github.com/ow2-proactive/scheduling/blob/0ce8cc469c23b4d53489c8b913c281664c91b6e0/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/LiveJobs.java#L848): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. ![MINOR][MINOR] [LiveJobs.java#L884](https://github.com/ow2-proactive/scheduling/blob/0ce8cc469c23b4d53489c8b913c281664c91b6e0/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/LiveJobs.java#L884): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. ![MINOR][MINOR] [SchedulingService.java#L628](https://github.com/ow2-proactive/scheduling/blob/0ce8cc469c23b4d53489c8b913c281664c91b6e0/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulingService.java#L628): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. ![MINOR][MINOR] [SchedulingService.java#L660](https://github.com/ow2-proactive/scheduling/blob/0ce8cc469c23b4d53489c8b913c281664c91b6e0/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulingService.java#L660): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. ![MINOR][MINOR] [SchedulingService.java#L692](https://github.com/ow2-proactive/scheduling/blob/0ce8cc469c23b4d53489c8b913c281664c91b6e0/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulingService.java#L692): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. ![MINOR][MINOR] [SchedulingService.java#L722](https://github.com/ow2-proactive/scheduling/blob/0ce8cc469c23b4d53489c8b913c281664c91b6e0/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulingService.java#L722): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. ![MINOR][MINOR] [SchedulingService.java#L750](https://github.com/ow2-proactive/scheduling/blob/0ce8cc469c23b4d53489c8b913c281664c91b6e0/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulingService.java#L750): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: scheduler/scheduler-server/src/test/java/functionaltests/rm/TestRMReconnectionWhileRunning.java [hunk 13]
```diff
@@ -0,0 +1,107 @@
...
+        schedulerHelper = new SchedulerTHelper(false, true);
+        schedulerHelper.createNodeSource("local", 3);
+        schedulerHelper.getSchedulerInterface().changePolicy("functionaltests.rm.PolicyWhichThrowsExceptions");
+        JobId jobId = schedulerHelper.submitJob(new File(runningJob.toURI()).getAbsolutePath());
+        JobId jobId2 = schedulerHelper.submitJob(new File(runningJob1.toURI()).getAbsolutePath());
+        JobId jobId3 = schedulerHelper.submitJob(new File(runningJob2.toURI()).getAbsolutePath());
+
+        schedulerHelper.waitForEventTaskRunning(jobId, "running_task_for20s");
+
+        Thread.sleep(200000); // sleep enough that all jobs should be able to finish
```
Comment (sonarqube-activeeon) [inline scheduler/scheduler-server/src/test/java/functionaltests/rm/TestRMReconnectionWhileRunning.java]: MAJOR Remove this use of "Thread.sleep()". [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2925)

**Comment by sonarqube-activeeon** [comment]
SonarQube analysis reported 19 issues
* ![CRITICAL][CRITICAL] 1 critical
* ![MAJOR][MAJOR] 1 major
* ![MINOR][MINOR] 17 minor

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [InternalJob.java#L399](https://github.com/ow2-proactive/scheduling/blob/3f382184b8bfc9d8905be489054de1eb781e8c35/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/job/InternalJob.java#L399): Add a default case to this switch. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ASwitchLastCaseIsDefaultCheck)
1. ![MINOR][MINOR] [LiveJobs.java#L705](https://github.com/ow2-proactive/scheduling/blob/3f382184b8bfc9d8905be489054de1eb781e8c35/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/LiveJobs.java#L705): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. ![MINOR][MINOR] [LiveJobs.java#L793](https://github.com/ow2-proactive/scheduling/blob/3f382184b8bfc9d8905be489054de1eb781e8c35/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/LiveJobs.java#L793): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. ![MINOR][MINOR] [LiveJobs.java#L848](https://github.com/ow2-proactive/scheduling/blob/3f382184b8bfc9d8905be489054de1eb781e8c35/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/LiveJobs.java#L848): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. ![MINOR][MINOR] [LiveJobs.java#L884](https://github.com/ow2-proactive/scheduling/blob/3f382184b8bfc9d8905be489054de1eb781e8c35/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/LiveJobs.java#L884): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. ![MINOR][MINOR] [SchedulingService.java#L628](https://github.com/ow2-proactive/scheduling/blob/3f382184b8bfc9d8905be489054de1eb781e8c35/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulingService.java#L628): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. ![MINOR][MINOR] [SchedulingService.java#L660](https://github.com/ow2-proactive/scheduling/blob/3f382184b8bfc9d8905be489054de1eb781e8c35/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulingService.java#L660): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. ![MINOR][MINOR] [SchedulingService.java#L692](https://github.com/ow2-proactive/scheduling/blob/3f382184b8bfc9d8905be489054de1eb781e8c35/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulingService.java#L692): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. ![MINOR][MINOR] [SchedulingService.java#L722](https://github.com/ow2-proactive/scheduling/blob/3f382184b8bfc9d8905be489054de1eb781e8c35/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulingService.java#L722): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. ![MINOR][MINOR] [SchedulingService.java#L750](https://github.com/ow2-proactive/scheduling/blob/3f382184b8bfc9d8905be489054de1eb781e8c35/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulingService.java#L750): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by sonarqube-activeeon** [comment]
SonarQube analysis reported 19 issues
* ![CRITICAL][CRITICAL] 1 critical
* ![MAJOR][MAJOR] 1 major
* ![MINOR][MINOR] 17 minor

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [InternalJob.java#L399](https://github.com/ow2-proactive/scheduling/blob/dc7495d9601e3eb0fd623f53826b6b0f10405791/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/job/InternalJob.java#L399): Add a default case to this switch. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ASwitchLastCaseIsDefaultCheck)
1. ![MINOR][MINOR] [LiveJobs.java#L705](https://github.com/ow2-proactive/scheduling/blob/dc7495d9601e3eb0fd623f53826b6b0f10405791/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/LiveJobs.java#L705): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. ![MINOR][MINOR] [LiveJobs.java#L793](https://github.com/ow2-proactive/scheduling/blob/dc7495d9601e3eb0fd623f53826b6b0f10405791/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/LiveJobs.java#L793): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. ![MINOR][MINOR] [LiveJobs.java#L848](https://github.com/ow2-proactive/scheduling/blob/dc7495d9601e3eb0fd623f53826b6b0f10405791/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/LiveJobs.java#L848): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. ![MINOR][MINOR] [LiveJobs.java#L884](https://github.com/ow2-proactive/scheduling/blob/dc7495d9601e3eb0fd623f53826b6b0f10405791/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/LiveJobs.java#L884): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. ![MINOR][MINOR] [SchedulingService.java#L628](https://github.com/ow2-proactive/scheduling/blob/dc7495d9601e3eb0fd623f53826b6b0f10405791/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulingService.java#L628): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. ![MINOR][MINOR] [SchedulingService.java#L660](https://github.com/ow2-proactive/scheduling/blob/dc7495d9601e3eb0fd623f53826b6b0f10405791/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulingService.java#L660): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. ![MINOR][MINOR] [SchedulingService.java#L692](https://github.com/ow2-proactive/scheduling/blob/dc7495d9601e3eb0fd623f53826b6b0f10405791/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulingService.java#L692): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. ![MINOR][MINOR] [SchedulingService.java#L722](https://github.com/ow2-proactive/scheduling/blob/dc7495d9601e3eb0fd623f53826b6b0f10405791/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulingService.java#L722): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. ![MINOR][MINOR] [SchedulingService.java#L750](https://github.com/ow2-proactive/scheduling/blob/dc7495d9601e3eb0fd623f53826b6b0f10405791/scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulingService.java#L750): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request closed without merging**